File: Training/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if (x < 0)
            return false;
        
        int64_t x_copy = x, reverse = 0;  
        while (x_copy) {
            reverse = reverse * 10 + x_copy % 10;
            x_copy /= 10;
        }

        if (reverse == x)
            return true;

        return false;
    }
};

File: Training/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        cast = str(x)
        end = len(cast) - 1
        half_len = int(len(cast) / 2)
        for i in range(half_len):
            if cast[i] != cast[end]:
                return False
            end -= 1
        return True
